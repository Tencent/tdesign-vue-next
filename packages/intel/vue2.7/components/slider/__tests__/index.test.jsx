import { mount } from '@vue/test-utils';
import Vue from 'vue';
import Slider from '@/src/slider/index.ts';
import TInputNumber from '@/src/input-number/index.ts';

// every component needs four parts: props/events/slots/functions.
describe('Slider', () => {
  // test props api
  describe(':props', () => {
    it('', () => {
      const wrapper = mount({
        render() {
          return <Slider></Slider>;
        },
      });
      // 写入快照
      expect(wrapper.element).toMatchSnapshot();
    });
    it('disabled', () => {
      const wrapper = mount({
        render() {
          return <Slider disabled={true}></Slider>;
        },
      });
      expect(wrapper.find('.t-is-disabled').exists()).toBe(true);
    });
    it('inputNumberProps', () => {
      const inputNumberPropsNormalWrapper = mount({
        render() {
          return <Slider inputNumberProps></Slider>;
        },
      });
      expect(inputNumberPropsNormalWrapper.find('.t-input-number').exists()).toBe(true);
      expect(inputNumberPropsNormalWrapper.find('.t-input-number--column').exists()).toBe(true);
      const inputNumberPropsRowWrapper = mount({
        render() {
          return <Slider inputNumberProps={{ theme: 'row' }}></Slider>;
        },
      });
      expect(inputNumberPropsRowWrapper.find('.t-input-number--row').exists()).toBe(true);
    });
    it('max and min', () => {
      const wrapper = mount({
        render() {
          return <Slider min={10} max={50}></Slider>;
        },
      });
      expect(wrapper.find('.t-slider').attributes('aria-valuemax')).toBe('50');
      expect(wrapper.find('.t-slider').attributes('aria-valuemin')).toBe('10');
    });
    it('layout', () => {
      const wrapper = mount({
        render() {
          return <Slider layout="vertical"></Slider>;
        },
      });
      // 配置layout属性为vertical，且rail的样式为垂直方向 height 100%
      expect(wrapper.find('.t-slider--vertical').exists()).toBe(true);
      expect(wrapper.find('.t-slider__rail').attributes('style')).toBe('height: 100%;');
    });
    it('range', () => {
      const wrapper = mount({
        render() {
          return <Slider layout="vertical" range></Slider>;
        },
      });
      // 配置range属性 存在两个游标button
      expect(wrapper.findAll('.t-slider__button').length).toBe(2);
    });
  });
});

// 测试Slider下的marks传参模板
describe('Slider [marks]', () => {
  const wrapper = mount(Slider, {
    propsData: {
      inputNumberProps: { theme: 'column' },
      value: 2,
      marks: {
        0: '0°C',
        8: '8°C',
        12: '12°C',
        37: '37°C',
        // 这里单元测试不支持传递node节点
        // 50: <strong style="color: #1989FA">50°C</strong>,
        // 60: <button style="color: #1989FA">60°C</button>,
      },
    },
  });
  // 渲染是否正确
  it('render right', () => {
    expect(wrapper.html()).toContain('aria-valuetext="2"');
    expect(wrapper.html()).toContain('37°C');
    expect(wrapper.element).toMatchSnapshot();
  });

  // 输入框输入20
  it('change value on input-number', (done) => {
    // console.log(wrapper.findComponent(TInputNumber).vm);
    Vue.nextTick(() => {
      wrapper.findComponent(TInputNumber).vm.$emit('change', 20);
      done();
    });
  });

  // 检查内部值是否变化
  it('check change value', () => {
    expect(wrapper.vm.prevValue).toBe(20);
  });

  // 检查内部函数是否返回正确
  it('call setValues()', () => {
    expect(wrapper.vm.setValues(1000)).toBe(100);
  });
});

describe('Slider [vertical-marks]', () => {
  const wrapper = mount(Slider, {
    propsData: {
      value: [30, 70],
      layout: 'vertical',
      range: true,
      marks: {
        0: '0°C',
        8: '8°C',
        12: '12°C',
        37: '37°C',
        // 这里单元测试不支持传递node节点
        // 50: <strong style="color: #1989FA">50°C</strong>,
        // 60: <button style="color: #1989FA">60°C</button>,
      },
    },
  });
  // 渲染是否正确
  it('render right', () => {
    expect(wrapper.html()).toContain('aria-valuetext="30-70"');
    expect(wrapper.html()).toContain('37°C');
    expect(wrapper.element).toMatchSnapshot();
  });
  // 检查内部函数是否返回正确
  it('call setValues()', () => {
    expect(wrapper.vm.setValues([300, 200])).toEqual([30, 100]);
    expect(wrapper.vm.setValues([-200, -300])).toEqual([0, 100]);
  });
});
