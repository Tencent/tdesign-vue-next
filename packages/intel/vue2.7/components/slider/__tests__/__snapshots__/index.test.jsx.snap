// Vitest Snapshot v1

exports[`Slider > :props >  1`] = `
<div
  aria-valuetext="0"
  class="t-slider__container"
>
  <div
    aria-orientation="horizontal"
    aria-valuemax="100"
    aria-valuemin="0"
    class="t-slider"
    role="slider"
  >
    <div
      class="t-slider__rail"
    >
      <div
        class="t-slider__track"
        style="width: 0%; left: 0%;"
      />
      <div
        class="t-slider__button-wrapper"
        show-tooltip="true"
        style="left: 0%;"
        tabindex="0"
      >
        <div
          class="t-slider__button"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Slider [marks] > render right 1`] = `
<div
  aria-valuetext="2"
  class="t-slider__container"
>
  <div
    aria-orientation="horizontal"
    aria-valuemax="100"
    aria-valuemin="0"
    class="t-slider t-slider--with-input"
    role="slider"
  >
    <div
      class="t-slider__rail show-input"
    >
      <div
        class="t-slider__track"
        style="width: 2%; left: 0%;"
      />
      <div
        class="t-slider__button-wrapper"
        show-tooltip="true"
        style="left: 2%;"
        tabindex="0"
      >
        <div
          class="t-slider__button"
        />
      </div>
      <div>
        <div
          class="t-slider__stops"
        >
          <div
            class="t-slider__stop t-slider__mark-stop"
            style="left: 8%;"
          />
          <div
            class="t-slider__stop t-slider__mark-stop"
            style="left: 12%;"
          />
          <div
            class="t-slider__stop t-slider__mark-stop"
            style="left: 37%;"
          />
        </div>
        <div
          class="t-slider__mark"
        >
          <div
            class="t-slider__mark-text"
            style="left: 0%;"
          >
            0°C
          </div>
          <div
            class="t-slider__mark-text"
            style="left: 8%;"
          >
            8°C
          </div>
          <div
            class="t-slider__mark-text"
            style="left: 12%;"
          >
            12°C
          </div>
          <div
            class="t-slider__mark-text"
            style="left: 37%;"
          >
            37°C
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="t-slider__input-container"
  >
    <div
      class="t-input-number t-size-m t-is-controls-right t-input-number--column t-slider__input"
    >
      <button
        class="t-button t-button--variant-outline t-button--theme-default t-button--shape-square t-input-number__decrease"
        type="button"
      >
        <svg
          class="t-icon t-icon-chevron-down t-size-m"
          fill="none"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M17.5 8.09l-5.5 5.5-5.5-5.5L5.09 9.5 12 16.41l6.91-6.91-1.41-1.41z"
            fill="currentColor"
          />
        </svg>
      </button>
      <div
        class="t-input__wrap"
      >
        <div
          class="t-input"
        >
          <input
            autocomplete="off"
            class="t-input__inner"
            placeholder=""
            type="text"
            unselectable="off"
          />
        </div>
      </div>
      <button
        class="t-button t-button--variant-outline t-button--theme-default t-button--shape-square t-input-number__increase"
        type="button"
      >
        <svg
          class="t-icon t-icon-chevron-up t-size-m"
          fill="none"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M17.5 15.91l-5.5-5.5-5.5 5.5-1.41-1.41L12 7.59l6.91 6.91-1.41 1.41z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`Slider [vertical-marks] > render right 1`] = `
<div
  aria-valuetext="30-70"
  class="t-slider__container is-vertical"
>
  <div
    aria-orientation="vertical"
    aria-valuemax="100"
    aria-valuemin="0"
    class="t-slider is-vertical t-slider--vertical"
    role="slider"
  >
    <div
      class="t-slider__rail"
      style="height: 100%;"
    >
      <div
        class="t-slider__track"
        style="height: 40%; bottom: 30%;"
      />
      <div
        class="t-slider__button-wrapper"
        show-tooltip="true"
        style="bottom: 30%;"
        tabindex="0"
      >
        <div
          class="t-slider__button"
        />
      </div>
      <div
        class="t-slider__button-wrapper"
        show-tooltip="true"
        style="bottom: 70%;"
        tabindex="0"
      >
        <div
          class="t-slider__button"
        />
      </div>
      <div>
        <div
          class="t-slider__stops"
        >
          <div
            class="t-slider__stop t-slider__mark-stop"
            style="top: calc(92% - 1px);"
          />
          <div
            class="t-slider__stop t-slider__mark-stop"
            style="top: calc(88% - 1px);"
          />
          <div
            class="t-slider__stop t-slider__mark-stop"
            style="top: calc(63% - 1px);"
          />
        </div>
        <div
          class="t-slider__mark"
        >
          <div
            class="t-slider__mark-text"
            style="top: calc(100% - 1px);"
          >
            0°C
          </div>
          <div
            class="t-slider__mark-text"
            style="top: calc(92% - 1px);"
          >
            8°C
          </div>
          <div
            class="t-slider__mark-text"
            style="top: calc(88% - 1px);"
          >
            12°C
          </div>
          <div
            class="t-slider__mark-text"
            style="top: calc(63% - 1px);"
          >
            37°C
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
